src/e2e/puppeteer/helpers/mousePointerHelper.ts

```diff
@@ -0,0 +1,87 @@
+import { Page } from 'puppeteer'
+
+/**
+ * Installs mouse helper on the page to display the mouse pointer.
+ */
+async function installMouseHelper(page: Page) {
+  await page.evaluate(() => {
+    // Install mouse helper only for top-level frame.
+    if (window !== window.parent) return
+
+    const box = document.createElement('puppeteer-mouse-pointer')
+    const styleElement = document.createElement('style')
+    styleElement.innerHTML = `
+      puppeteer-mouse-pointer {
+        pointer-events: none;
+        position: absolute;
+        top: 0;
+        z-index: 10000;
+        left: 0;
+        width: 30px;
+        height: 30px;
+        background-image: url('https://cursor.in/assets/cursor.svg');
+        background-size: cover;
+        border: none;
+        margin: -10px 0 0 -10px;
+        padding: 0;
+        transition: background .2s, border-radius .2s, border-color .2s;
+      }
+      puppeteer-mouse-pointer.button-1 {
+        transition: none;
+        background: rgba(0,0,0,0.9);
+      }
```

Comment:
I think the button logic is part of the original circular pointer, and is no longer relevant.